product_contract/models/sale_order.py

```diff
@@ -71,11 +71,11 @@ def action_create_contract(self):
                 )
             )
             for contract_template in line_to_create_contract.mapped(
-                'product_id.contract_template_id'
+                'product_id.property_contract_template_id'
             ):
                 order_lines = line_to_create_contract.filtered(
```

Comment:
It would be safer to add `with_context(force_company=line_to_create_contract.company_id)` for being sure the contract template is got from the right company. What awful long variable names by the way...